app/knights.py

```diff
@@ -12,8 +12,7 @@ def __init__(self, params: dict) -> None:
         self.protection = 0
 
     def apply_armor(self) -> None:
-        for armor in self.armour:
-            self.protection += armor["protection"]
+        self.protection = sum([armor["protection"] for armor in self.armour])

```

Comment:
We can use simple generator for this. Even without list

`self.protection = sum(armor["protection"] for armor in self.armour)`

Reply:
makes sense🫠